lib/App.js

```diff
@@ -1107,17 +1079,29 @@ class App {
     });
   }
 
-  async _getPackFileEntries(prunePaths) {
+  async _getAppSourceFiles() {
+    const DEFAULT_IGNORE_RULES_FILE = '$default-ignore-rules';
+
     const walker = new ignoreWalk.Walker({
       path: this.path,
-      ignoreFiles: ['.homeyignore', ''], // Empty string is workaround for adding default ignore rules
+      ignoreFiles: ['.homeyignore', DEFAULT_IGNORE_RULES_FILE],
       includeEmpty: true,
       follow: true,
     });
 
-    // Add default ignore rules for dotfiles, env.json and npm prune paths
-    const ignoreRules = ['.*', '*.ts', 'tsconfig.json', 'env.json', ...prunePaths];
-    walker.onReadIgnoreFile('', ignoreRules.join('\r\n'), () => {});
+    const prunePaths = await NpmCommands.getPrunePaths({ path: this.path });
+
+    const ignoreRules = [
+      '.*',
+      '*.ts',
+      'tsconfig.json',
+      'env.json',
+      '*.compose.json',
+      'build/*',
```

Comment:
`*.compose.json` and `build/*` where added to the ignore rules. We run compose before copying the "source" files so we don't need the compose files in the `build/` anymore. We obviously don't want to copy `build/` into `build/`